test/instance_unittest.cc

```diff
@@ -0,0 +1,77 @@
+#include "../src/configuration.h"
+#include "../src/environment.h"
+#include "../src/instance.h"
+#include "../src/resource.h"
+#include "../src/updater.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+namespace {
+
+class InstanceTest : public ::testing::Test {
+ protected:
+  const MetadataStore::Metadata&& GetResourceMetadata(
+      const MetadataUpdater::ResourceMetadata* rm) {
+    return std::move(rm->metadata);
+  }
+
+  MonitoredResource GetMonitoredResource(
+      const MetadataUpdater::ResourceMetadata* rm) {
+    return rm->resource;
+  }
+
+  std::vector<std::string> GetResourceIds(
+      const MetadataUpdater::ResourceMetadata* rm) {
+    return rm->ids;
+  }
+};
+
+TEST_F(InstanceTest, GetInstanceMonitoredResource) {
+  Configuration config(std::istringstream(
+      "InstanceResourceType: gce_instance\n"
+      "InstanceId: 1234567891011\n"
+      "InstanceZone: us-east1-b\n"
+  ));
+  Environment env(config);
+  MonitoredResource mr_actual =
+      InstanceReader::InstanceResource(env);
+  MonitoredResource mr_expected(
+      "gce_instance",
+      {{"instance_id", "1234567891011"}, {"zone", "us-east1-b"}});
+  EXPECT_EQ(mr_expected, mr_actual);
+}
+
+TEST_F(InstanceTest, GetInstanceMonitoredResourceWithEmptyConfig) {
+  Environment env(*(new Configuration()));
+  MonitoredResource mr_actual =
+      InstanceReader::InstanceResource(env);
+  MonitoredResource mr_expected(
+      "gce_instance", {{"instance_id", ""}, {"zone", ""}});
+  EXPECT_EQ(mr_expected, mr_actual);
+}
+
+TEST_F(InstanceTest, GetInstanceMetatadataQuery) {
+  Configuration config(std::istringstream(
+      "InstanceResourceType: gce_instance\n"
+      "InstanceId: 1234567891011\n"
+      "InstanceZone: us-east1-b\n"
+  ));
+  InstanceReader reader(config);
+  const auto& result = reader.MetadataQuery();
+  MonitoredResource mr_expected(
+      "gce_instance",
+      {{"instance_id", "1234567891011"}, {"zone", "us-east1-b"}});
+  std::string vinit[] = {"", "1234567891011"};
+  std::vector<std::string> ids_exp = {vinit, std::end(vinit)};
```

Comment:
```cpp
const std::vector<std:string> ids_expected{"", "1234567891011};
```

Reply:
The parentheses were optional, BTW.
I also suggested renaming it to `ids_expected`, but now that you've inlined the monitored resource, might as well inline this too:
```cpp
  EXPECT_EQ(std::vector<std::string>{"", "1234567891011"}, GetResourceIds(rm));
```
Don't have an easy way to test this now, but it might even auto-convert to a vector:
```cpp
  EXPECT_EQ({"", "1234567891011"}, GetResourceIds(rm));
```

Reply:
Done. Sorry I missed the rename. Also inline auto-convert doesn't work actually.

Reply:
Auto-convert doesn't, but inlining should.

Reply:
Done.